### PR TITLE
Feat: login feature

### LINKED ISSUE #9: [LOGIN] add login

### PR DESCRIPTION
**Pull request by thatguycodes**
* integrate login form with server-side endpoint
* store token in local storage
* add interceptor

### CODE REVIEW

**Comment by MwinyiMoha** [comment]
@MrObonye I pushed the `ngbootstrap` files just in case

File: src/app/auth/login/login.component.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<body class="text-center">
```
Comment (MwinyiMoha) [inline src/app/auth/login/login.component.html]: Shouldn't we use a div here?? Considering the entire app is rendered within the `body` tag of `index.html`. Feels like a body tag within a body tag

**Review by MwinyiMoha — DISMISSED** [review]
Good work here @MrObonye !!

File: src/app/auth/login/login.component.html (hunk 1 above)
Comment (thatguycodes) [reply src/app/auth/login/login.component.html]: yep should be a div

**Review by MwinyiMoha — APPROVED** [review]
Looks good!!

**Pull request merged**